moto/apigateway/responses.py

```diff
@@ -368,26 +369,21 @@ def stages(self, request, full_url, headers):
         function_id = url_path_parts[2]
         stage_name = url_path_parts[4]
 
-        if self.method == "GET":
-            try:
+        try:
+            if self.method == "GET":
                 stage_response = self.backend.get_stage(function_id, stage_name)
-            except StageNotFoundException as error:
-                return (
-                    error.code,
-                    {},
-                    '{{"message":"{0}","code":"{1}"}}'.format(
-                        error.message, error.error_type
-                    ),
+
+            elif self.method == "PATCH":
+                patch_operations = self._get_param("patchOperations")
+                stage_response = self.backend.update_stage(
+                    function_id, stage_name, patch_operations
                 )
-        elif self.method == "PATCH":
-            patch_operations = self._get_param("patchOperations")
-            stage_response = self.backend.update_stage(
-                function_id, stage_name, patch_operations
-            )
-        elif self.method == "DELETE":
-            self.backend.delete_stage(function_id, stage_name)
-            return 202, {}, "{}"
-        return 200, {}, json.dumps(stage_response)
+            elif self.method == "DELETE":
+                self.backend.delete_stage(function_id, stage_name)
+                return 202, {}, "{}"
+            return 200, {}, json.dumps(stage_response)
+        except StageNotFoundException as error:
+            return error.code, {}, error.get_body()
```

Comment:
The models throw some exceptions, but they weren't properly handled yet, resulting in a wrong error coming from boto3